File: common/src/main/java/com/ultreon/mods/advanceddebug/api/common/IntSize.java
package com.ultreon.mods.advanceddebug.api.common;

import com.ultreon.mods.advanceddebug.api.client.formatter.IFormatterContext;

import java.util.Objects;

public final class IntSize implements IFormattable {
    public int width;
    public int height;

    public IntSize(int width, int height) {
        this.width = width;
        this.height = height;
    }

    public int getWidth() {
        return this.width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return this.height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    @Override
    public void format(IFormatterContext ctx) {
        ctx.number(this.width).operator(" × ").number(this.height);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || this.getClass() != o.getClass()) return false;
        IntSize size = (IntSize) o;
        return this.width == size.width && this.height == size.height;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.width, this.height);
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/api/common/IntSize.kt
package com.ultreon.mods.advanceddebug.api.common

import com.ultreon.mods.advanceddebug.api.client.formatter.IFormatterContext
import java.util.*

class IntSize(var width: Int, var height: Int) : IFormattable {

    override fun format(ctx: IFormatterContext) {
        ctx.number(width).operator(" × ").number(height)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val size = other as IntSize
        return width == size.width && height == size.height
    }

    override fun hashCode(): Int {
        return Objects.hash(width, height)
    }
}

val Int.squared get() = IntSize(this, this)
fun size(width: Int, height: Int) = IntSize(width, height)
